src/devices/nm-device-6lowpan.c

```diff
@@ -0,0 +1,371 @@
+	                           NULL,
+	                           _("6LOWPAN connection"),
+	                           NULL,
+	                           TRUE);
+
+	s_6lowpan = nm_connection_get_setting_6lowpan (connection);
+	if (!s_6lowpan) {
+		g_set_error_literal (error, NM_DEVICE_ERROR, NM_DEVICE_ERROR_INVALID_CONNECTION,
+		                     "A '6lowpan' setting is required.");
+		return FALSE;
+	}
+
+	/* If there's no 6LoWPAN interface, no parent, and no hardware address in the
+	 * settings, then there's not enough information to complete the setting.
+	 */
+	if (   !nm_setting_6lowpan_get_parent (s_6lowpan)
+	    && !nm_device_match_hwaddr (device, connection, TRUE)) {
+		g_set_error_literal (error, NM_DEVICE_ERROR, NM_DEVICE_ERROR_INVALID_CONNECTION,
+		                     "The '6lowpan' setting had no interface name, parent, or hardware address.");
+		return FALSE;
+	}
+
+	return TRUE;
+}
+
+static void
+update_connection (NMDevice *device, NMConnection *connection)
+{
+	NMSetting6Lowpan *s_6lowpan = nm_connection_get_setting_6lowpan (connection);
+	NMDevice *parent_device;
+	const char *setting_parent, *new_parent;
+
+	if (!s_6lowpan) {
+		s_6lowpan = (NMSetting6Lowpan *) nm_setting_6lowpan_new ();
+		nm_connection_add_setting (connection, (NMSetting *) s_6lowpan);
+	}
+
+	/* Update parent in the connection; default to parent's interface name */
+	parent_device = nm_device_parent_get_device (device);
+	if (parent_device) {
+		new_parent = nm_device_get_iface (parent_device);
+		setting_parent = nm_setting_6lowpan_get_parent (s_6lowpan);
+		if (setting_parent && nm_utils_is_uuid (setting_parent)) {
+			NMConnection *parent_connection;
+
+			/* Don't change a parent specified by UUID if it's still valid */
+			parent_connection = (NMConnection *) nm_settings_get_connection_by_uuid (nm_device_get_settings (device), setting_parent);
+			if (parent_connection && nm_device_check_connection_compatible (parent_device, parent_connection))
+				new_parent = NULL;
+		}
+		if (new_parent)
+			g_object_set (s_6lowpan, NM_SETTING_6LOWPAN_PARENT, new_parent, NULL);
+	} else
+		g_object_set (s_6lowpan, NM_SETTING_6LOWPAN_PARENT, NULL, NULL);
+
+}
+
+static NMActStageReturn
+act_stage1_prepare (NMDevice *dev, NMDeviceStateReason *out_failure_reason)
+{
+	NMActStageReturn ret;
+
+	ret = NM_DEVICE_CLASS (nm_device_6lowpan_parent_class)->act_stage1_prepare (dev, out_failure_reason);
+	if (ret != NM_ACT_STAGE_RETURN_SUCCESS)
+		return ret;
+
+	if (!nm_device_hw_addr_set_cloned (dev, nm_device_get_applied_connection (dev), FALSE))
+		return NM_ACT_STAGE_RETURN_FAILURE;
+	return NM_ACT_STAGE_RETURN_SUCCESS;
+}
+
+/*****************************************************************************/
+
+static void
+nm_device_6lowpan_init (NMDevice6Lowpan *self)
+{
+}
+
+static const NMDBusInterfaceInfoExtended interface_info_device_6lowpan = {
+	.parent = NM_DEFINE_GDBUS_INTERFACE_INFO_INIT (
+		NM_DBUS_INTERFACE_DEVICE_6LOWPAN,
```

Comment:
Lacks D-Bus Property "Parent". Don't make it a legacy property (meaning: don't use `NM_DEFINE_DBUS_PROPERTY_INFO_EXTENDED_READABLE_L`).